cwl/src/main/scala/cwl/CwlWomExpression.scala

```diff
@@ -17,6 +19,23 @@ sealed trait CwlWomExpression extends WomExpression {
   override def evaluateType(inputTypes: Map[String, WomType]): ErrorOr[WomType] = cwlExpressionType.validNel
 }
 
+case class JobPreparationExpression(expression: Expression,
+                                    override val inputs: Set[String]) extends CwlWomExpression {
+  val cwlExpressionType = WomAnyType
+
+  override def sourceString = expression match {
+    case Expression.ECMAScriptExpression(s) => s.value
+    case Expression.ECMAScriptFunction(s) => s.value
+  }
+
+  override def evaluateValue(inputValues: Map[String, WomValue], ioFunctionSet: IoFunctionSet) = {
+    val pc = ParameterContext().withInputs(inputValues, ioFunctionSet)
+    Try(expression.fold(EvaluateExpression).apply(pc)).toErrorOr
```

Comment:
ToL  we should return a try from `EvaluateExpression` as opposed to having to protect ourselves.